packages/datagateway-search/src/searchBoxContainer.component.test.tsx

```diff
@@ -14,41 +19,66 @@ jest.mock('react-redux', () => ({
 }));
 
 describe('SearchBoxContainer - Tests', () => {
-  const createWrapper = (): ShallowWrapper => {
-    return shallow(
-      <SearchBoxContainer
-        initiateSearch={jest.fn()}
-        onSearchTextChange={jest.fn()}
-        searchText=""
-      />
+  function renderComponent({ initialState }): RenderResult {
+    return render(
+      <Provider store={configureStore([thunk])(initialState)}>
+        <MemoryRouter>
+          <SearchBoxContainer
+            initiateSearch={jest.fn()}
+            onSearchTextChange={jest.fn()}
+            searchText=""
+          />
+        </MemoryRouter>
+      </Provider>
     );
-  };
+  }
+
+  let state: DeepPartial<StateType>;
 
   beforeEach(() => {
-    useSelector.mockImplementation(() => {
-      return initialState;
-    });
+    state = {
+      dgsearch: {
+        sideLayout: true,
+        searchableEntities: [],
+      },
+    };
   });
 
   it('renders searchBoxContainer correctly', () => {
-    const wrapper = createWrapper();
-    expect(wrapper).toMatchSnapshot();
+    const { asFragment } = renderComponent({ initialState: state });
+    expect(asFragment()).toMatchSnapshot();
   });
 });
```

Comment:
We should add more tests to check for things like the MyDataCheckbox appearing if the user is not logged in  anonymously, if it calls the correct method, etc.